src/shared/actions/entries.js

```diff
@@ -20,7 +20,27 @@ import {
 import { loadOrUnlockArchive } from '../../shared/actions/archives';
 import { loadGroup } from '../../shared/actions/groups';
 
-export const selectEntry = createAction(ENTRIES_SELECTED);
+export const selectEntry = (
+  entryId,
+  currentEntry,
+  currentEntryMode
+) => async dispatch => {
+  try {
+    !currentEntry && currentEntryMode === 'new'
```

Comment:
You can get `currentEntry` and `currentEntryMode` here directly using `getState` and the selector functions, no need to pass them in from the UI component.